### PR TITLE
Module/instance thread rework

### PR DESCRIPTION
**Pull request by coolreader18**
# Description of Changes

# API and ABI

 - [ ] This is a breaking change to the module ABI
 - [ ] This is a breaking change to the module API
 - [ ] This is a breaking change to the ClientAPI
 - [ ] This is a breaking change to the SDK API

*If the API is breaking, please state below what will break*

### CODE REVIEW

File: crates/core/src/host/host_controller.rs [hunk 1]
```diff
@@ -216,36 +241,33 @@ impl HostController {
...
+        let module_host = match mhc.host_type {
             HostType::Wasmer => {
                 // make_actor with block_in_place since it's going to take some time to compute.
                 let start = Instant::now();
                 let actor = tokio::task::block_in_place(|| {
                     wasmer::make_actor(mhc.dbic, module_hash, &mhc.program_bytes, mhc.scheduler, energy_monitor)
                 })?;
-                log::trace!("wasmer::make_actor blocked for {}us", start.elapsed().as_micros());
-                ModuleHost::spawn(actor)
+                log::trace!("wasmer::make_actor blocked for {:?}", start.elapsed());
```
Comment (cloutiertyler) [inline crates/core/src/host/host_controller.rs:266]: Why is this debug now?

**Review by cloutiertyler — APPROVED** [review]
Please see my minor comment inline. 

NOTE: This *seems* much more complicated a system than we need. It feels overengineered to be when we could just `tokio::spawn` a task per reducer `call`, but I grant that might also have performance issues for long running compute. I think this is okay to merge iff you are willing to take responsibility for its performance in testnet and to add monitoring for the exact resource usages by this system including memory allocated by all the thread stacks.

File: crates/core/src/host/host_controller.rs (hunk 1 above)
Comment (coolreader18) [reply crates/core/src/host/host_controller.rs:266]: it does the same thing (when the Duration is in the microseconds), I didn't see a reason to do our own formatting when the Duration Debug impl works fine

**Comment by coolreader18** [comment]
What do you mean tokio::spawn per reducer call? Like, instead of the rayon thread pool?

**Comment by coolreader18** [comment]
Also the memory usage by thread stacks is constant - it's `2 * available_parallelism() * default_thread_stack_size (10MB I think on linux)`. Unless wasmer allocates separate wasm stack space on the heap, which it might, but it'd be no different from any other way of doing it.

**Pull request merged**